pom.xml

```diff
@@ -69,6 +69,11 @@
         <scope>import</scope>
         <type>pom</type>
       </dependency>
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>30.1-jre</version>
+      </dependency>
```

Comment:
this is likely not the fix you are looking for.
as the plugin is not using any special classloader it will not get these classes, but the classes from Jenkins core.
probably best at the moment to actual fix the reason why it is being pulled in (most likely the parent pom being old and containing a hpi-plugin version with a few known bugs when deciding what to bundle)

And if any transitive dependency ends up being newer that what is provided by the jenkins-core being bundled then pay special attention to that and work out what methods are being called and validate they are still compatable (included any potential serialized for of them!)

Reply:
I understand that we need to remove Guava, the problem being that it's used on the third party library. I'll see there what I can do..